spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -478,7 +558,10 @@ def stop_button_click_handler(self):
 
     def show_kernel_error(self, error):
         """Show kernel initialization errors in infowidget."""
-        InstallerIPythonKernelError(error)
+        self.error_text = error
+        if "issue1666807" not in error:

```

Comment:
This looks really ugly and it could lead to problems when building the installer for publishing it.

@mrclary, what do you think about this?

Reply:
@impact27, what is the significance of "issue1666807"? `InstallerIPythonKernelError` is perfectly safe to execute under any circumstance, so does not require flow control. `InstallerIPythonKernelError` will not do anything unless the installer test is running, i.e. the system environment variable `INSTALLER_TEST=1`; this only occurs on the GitHub server when explicitly testing the macOS application build.

Reply:
When building the installer, the work in this PR reveals that an error is printer relative to https://bugs.python.org/issue1666807. Basically python complains that some paths are relative and not absolute. I guess this is made to have a portable installer. I don't think the code here creates this error but rather reveals it. I will remove this but I think this will make the installer tests fail.

Reply:
Error in the `Create macOS App Bundle and DMG / macOS App Bundle (Lite)` test:
```
  This version of python seems to be incorrectly compiled
  (internal generated filenames are not absolute).
  This may make the debugger miss breakpoints.
  Related bug: http://bugs.python.org/issue1666807
```

Reply:
Okay, I've seen this happen recently. It is not related to anything in this PR and this PR did not uncover it. As far as this PR is concerned, that is good news. I was not aware, however, that it was causing the installer test to fail. AFAIK, this may not present an actual issue when using Spyder, but nevertheless needs to be addressed. I'll look into it right away.

Reply:
Okay, quick clarification. This warning printed in the IPython console began showing up recently, but did not cause the installer to fail. This PR _did_ begin causing the installer to fail; I assume that it was not captured before this PR. I don't know how malicious this warning is, but I'll try to find the root cause.

Reply:
I could add back the `if "issue1666807" not in error:` so the tests pass and add an issue to the issue tracker so this can be handled separately?

Reply:
Unfortunately, I think that is our only option right now. This issue has me stumped. I think we should reopen #16828 rather than creating a new issue.